pilot/pkg/proxy/envoy/v2/discovery.go

```diff
@@ -192,6 +194,28 @@ func (s *DiscoveryServer) Start(stopCh <-chan struct{}) {
 	go s.sendPushes(stopCh)
 }
 
+func (s *DiscoveryServer) getNonK8sRegistries() []serviceregistry.Instance {
+	var registries []serviceregistry.Instance
+	var nonK8sRegistries []serviceregistry.Instance
+
+	if agg, ok := s.Env.ServiceDiscovery.(*aggregate.Controller); ok {
+		registries = agg.GetRegistries()
+	} else {
+		registries = []serviceregistry.Instance{
+			serviceregistry.Simple{
+				ServiceDiscovery: s.Env.ServiceDiscovery,
+			},
+		}
+	}
+
+	for _, registry := range registries {
+		if registry.Provider() != serviceregistry.Kubernetes || registry.Provider() != serviceregistry.External {
```

Comment:
@howardjohn Basically this condition is changed now to include Service Entry also